### ISSUE TITLE
Microdata examples incorrect markup

### ISSUE THREAD

**Issue by AndySky21** [schemaorg/schemaorg#2937]
https://schema.org/price is just an example.

_Note that both RDFa and Microdata syntax allow the use of a "content=" attribute_

On what planet? Microdata just expects the use of `content` on the only element which natively supports it: the `<meta>` element.
The use of `<span>` element in the Microdata example is totally incorrect. HTML has a completely different way to markup data with a machine-equivalent content: the `<data>` element in conjunction with the `value` attribute. This is the model supported (and suggested) by Microdata spec.
Speaking of new elements, why embedding non-textual numeric content (namely in ratings) with `<meta>` element? There is <meter> for that purpose, it supports articulated value semantics, it's used for Microdata and it's also visually rendered in page.

This scenario deserves updating.

**Comment by WeaverStever** [comment]
The W3.org Nu HTML5 validator says No errors no warnings on the price microdata example.
https://validator.w3.org/nu/#textarea
I've found the Nu validator to be quite strict.

Just for my own sake, I changed the span element to a div and it still passes. In some cases an empty div is used as a placeholder on the page for the the data-* element for use with javascript.  I can imagine that the itemprop / content pair could be useful there.

The content=value is also quite useful when you have to display a GMT date-time in the microdata, but want to present a human readable data-time to the user in an Event.  

The meta / content name pair are used when you have to include the microdata, but you don't want the markup visible on your page.  Frankly, without the meta and link elements, microdata would be quite useless.  My guess is that these elements are preferred over style="display:none;"  in a visible element.

The schema "price" example passes the Rich snippets testing tool, however it has several missing properties.
https://search.google.com/test/rich-results?id=-XlgKf0h8xJiz7sZVvVYiA

Missing field "sku" (optional)
Missing field "brand" (optional)
No global identifier provided (e.g., gtin, mpn, isbn) (optional)

**offers**
Missing field "priceValidUntil" (optional)
Missing field "url" (optional)

**Comment by AndySky21** [comment]
> The W3.org Nu HTML5 validator says No errors no warnings on the price microdata example.

The W3 port of Nu validator is known for not being updated regularly, I wouldn't follow it by the letter.
It is quite possible that the error is not recognized because `content` is, in fact, allowed in RDFa syntax.
[https://validator.nu/](https://validator.nu/) is quite specific in flagging content attribute on span element as _"Attribute content not allowed on element span at this point."_, feel free to test.

> In some cases an empty div is used as a placeholder on the page for the the data-* element for use with javascript. I can imagine that the itemprop / content pair could be useful there.

I fail in seeing how `content` and dataset can be related.

> The content=value is also quite useful when you have to display a GMT date-time in the microdata, but want to present a human readable data-time to the user in an Event

That's what `<time>` element has been meant for, and Microdata recognizes the `datetime` attribute on it. Indeed 12 years of brainstorming over semantic markup have been a waste of time.

> The meta / content name pair are used when you have to include the microdata, but you don't want the markup visible on your page. Frankly, without the meta and link elements, microdata would be quite useless

`meta@content` is possibly the worst use of rich metadata we can do. Microdata and RDFa share the common goal of making **visible** content recognizable for rich snippets whenever possible. Using them on `<meta>` should be limited, e.g. to redundant content or values which are obvious from the context for a human user but not for the machine, because it allows for passing to rich snippets values different from the real ones. Of course this risk still exists in `data@value` model, but it's easier to recognize there.

 > The schema "price" example passes the Rich snippets testing tool

Shared errors are not half errors. It only means that another tool fails in following specs. Which makes the tool buggy, not the spec.

**Comment by WeaverStever** [comment]
@AndySky21 

I don't know what validator you are using, but it isn't WC3.

This validates just fine when in the correct wrapper.
https://validator.w3.org/nu/#file
~~~~
<div itemscope itemtype="https://schema.org/Product">
  <span itemprop="name" content="microdata"> </span>
</div>
~~~~

So in type Event, you are telling me that I have to display startDate and endDate as in UTC format to my readers (2019-07-01T10:00:00-05:00) if I want to use microdata?

When I could display it as such -- this is **visible** content
~~~~
<span itemprop="startDate" content="2019-07-01T10:00:00-05:00"> 
    Monday, July 1, 2019 at 10:00am
</span>
~~~~

I never claimed the **missing properties** in the Rich Snippet tools were errors.  The tool may still be a little buggy, but in reality, Google is the "spec", they fund this project, host the website and consume the vast majority of the data. 

The "spec" for product / price is clearly spelled out here, 
https://developers.google.com/search/docs/advanced/structured-data/product
very little attention is paid to the examples on  schema as they have become quite buggy.

**Comment by AndySky21** [comment]
> I have to display startDate and endDate as in UTC format to my readers

Of course not. Ever heard of the `<time>` element?
`<time datetime="2019-07-01T10:00:00-05:00">Monday, July 1, 2019 at 10:00am</time>`
Which is semantic and does not need using generic `<span>` as it would've been the case 20 years ago. As I said before, this is not exactly _visible_ content. It is rather how visible content and machine-readable equivalent should now be associated, when you need both.

> The "spec" for product / price is clearly spelled out here

This isn't exactly how it works.
Specs for Schemas and vocabularies come from schema.org
Vocabularies can be implemented via Microdata, RDFa, RDFa Lite, and there once was another way of doing that, Microformats (it is abandoned now, AFAIK).
Google consumes rich snippets, as other search engines do, as Facebook and other social networks do. If they claim to support a syntax, they should actually follow that.

 > I don't know what validator you are using, but it isn't WC3.

I don't know where you have been lately.
After having maintained HTML5 spec since its first publication, currently the latest source for valid HTML content is [https://html.spec.whatwg.org/](https://html.spec.whatwg.org/), as stated in [latest W3 HTML5 (now superseded) recommendation](https://www.w3.org/TR/html52/).
Now, the original project for HTML Validator has merged into [The Nu Html Checker (v.Nu)](https://github.com/validator/validator). This project, now on GitHub for bug tracking and collaboration, is also referenced in validator.w3.org/nu (you can see it in the middle of the page, section **Reporting issues**, link "raise an issue". So, what's the difference? At a certain point, _HTML Living Standard_ as maintained by WHATWG and _HTML5_ as maintained by W3C started taking separate ways. W3 HTML5 claimed to extensively test for vendor support and authors' consistency of use, so it showed a version of the spec devoid of excessively experimental features, while WHATWG kept looking at the future (to make silly examples, validator.w3.org/nu flagged the use of `<hgroup>` and `<menu>` tags as error, while they were part of WHATWG standard and therefore they validated in validator.nu. Now this separation is over, as W3 appears to endorse WHATWG version of HTML.
Thus said, there's no better validation than the one we can explain to ourselves.
Microdata specifications come from 2 sources:

- [HTML Microdata (W3 working draft, now outdated)](https://www.w3.org/TR/2017/WD-microdata-20170504/)
- [HTML Living Standard - Microdata](https://html.spec.whatwg.org/multipage/microdata.html#microdata), the relevant part of HTML standard.

See by yourself if you can find out any mention of a `content` attribute anywhere in the text other than `<meta>`. More so, see by yourself if that attribute exists in HTML spec (either W3 HTML5.2 or HTML LS). Unfortunately there isn't. As you probably know, `content` attribute is part of an entirely different standard, RDFa, which serves a similar purpose than Microdata, with a different serialization and different vocabulary attributes. RDFa is by itself a W3C recommendation, and it comes both in a complete, stand-alone syntax (See [RDFa Core 1.1](https://www.w3.org/TR/rdfa-core/) and [HTML + RDFa](https://www.w3.org/TR/html-rdfa/)) consisting of 8 specific attributes, and a [Lite version](https://www.w3.org/TR/rdfa-lite/) capable of supporting some native HTML elements and attributes (basically what Microdata does). The problem is, RDFa Core 1.1 supports a `content` attribute, while RDFa Lite 1.1 doesn't (it relies on HTML, but it doesn't even support some HTML5 tags). I hate using RDFa (Facebook used to support that, instead of Microdata, but now it consumes both) because it forces authors to specify some metadata **both** for HTML **and** for Structured Data. RDFa (in all its incarnations) used to be more reliable than Microdata, as the latter was only a Working Draft, but now that HTML LS is **the** standard HTML spec, I guess it's no longer the case).

This takes us to the final consideration: authors cannot produce content which is both valid HTML (don't care about validators, I mean standards) and valid, conforming and consumable Structured Data (in the Microdata flavor).

That is, until: 
- **either** Rich Snippets Validators start accepting the correct model (and perhaps it starts being correctly consumed by search engines and data mining tools) 
- **or** HTML/Microdata spec is updated to accept a `content` attribute

**Comment by WeaverStever** [comment]
If content= is not a domain wide element, where can I go to become an advocate for this? It's not rocket science, every element accepts title=. 
 
 I guess that name= was been depreciated a long time ago.  

Here, from the Living HTML5 living standard.
1.7.3 Extensibility
Authors can use the <meta name="" content=""> mechanism to include page-wide metadata.
https://html.spec.whatwg.org/

So your point about not using the meta / content pair in metadata doesn't hold water. (microdata must conform to html5, not vice versa).

There is a an example of an embedded meta content pair in the documentation.


5.1.2 There is no semantic difference, for instance, between the following two examples:
~~~~
<figure>
 <img src="castle.jpeg">
 <figcaption><span itemscope><span itemprop="name">The Castle</span></span> (1986)</figcaption>
</figure>
~~~~

~~~~
<span itemscope><meta itemprop="name" content="The Castle"></span>
<figure>
 <img src="castle.jpeg">
 <figcaption>The Castle (1986)</figcaption>
</figure> 
~~~~

In the first example, why the itemscope needs its own span is beyond me. (My understanding is that you only need an itemscope when you are declaring a type.)  Question, is it itemscope? itemscope=? or itemscope=""?  I see all three flavors out there.

In the second example the meta section is unneeded and could be contained within the span -- poor example for what we are discussing, but from the examples.


There is nothing official about the Nu validator you are presenting.

From one of the author's bio:
 About Henri Sivonen
 [...]
"Previously, I developed **an** HTML5 validator and represented Mozilla on the HTML Working Group of the W3C. (**Note that since Open Source work often involves publishing personal opinions about work topics, you should not consider anything on this site as a position of Mozilla even if it carries Mozilla’s copyright notice**.)"
 
 The github project you reference is for the W3 validator.
 "It enables you to batch-check documents from the command line and from other scripts/apps, and to deploy your own instance of the checker as a service (like **validator.w3.org/nu)**. Its source code is available, as are instructions on how to build, test, and run the code."
 
Anyway, if we can't do shorthand, and avoid stuff like adding an extra span just to contain an empty itemscope, microdata will fail adoption because it makes the html unreadable and impossible to maintain.

**Comment by AndySky21** [comment]
You do know what "page-wide metadata" means, right? `<meta name=author>` has nothing to do with microdata. It's really basic HTML as it was used even in 4.01 version. 

The GitHub project IS validator.nu. They're the very same thing. W3 validator is a fork of that.

My point is that we need a reliable way to associate human- and machine-readable content in all those cases where both exist. Modern HTML offer appropriate tools, but all user agents consuming Microdata are supposed to respect Microdata specs and parse them correctly.
Namely,
- interpret `datetime` on `<time>` element for ISO date while keeping the human version intact (text content of time element)
- interpret `value` on `<data>` element when offering a machine-readable alternative for every other kind of data (text content of data element)
- interpret `value` on (empty) `<meter>` element for visually rendered numeric, meter-like values
- interpret `content` on `<meta>` element with `itemprop` attribute in residual cases when human-readable value is not necessary

It's really _this_ simple.

**Comment by WeaverStever** [comment]
The two online validators are not the same.

This passes on W3.org and fails on Validator.nu
I've known about WC3 for twenty some years, never heard of validator.nu

~~~~
<div itemscope itemtype="https://schema.org/Product">
  <span itemprop="name" content="microdata"> </span>
</div>
~~~~

**Comment by WeaverStever** [comment]
So referring back to your original post. 

You would rather see this?
~~~~
<span itemscope><meta itemprop="priceCurrency" content="USD" />$</span>
~~~~

than this?
~~~~
<span itemprop="priceCurrency" content="USD">$</span>
~~~~

Can you give an example of the data element?

**Comment by WeaverStever** [comment]
Nevermind, I see it.

This passes the W3 validator
~~~~ 
<div itemscope itemtype="http://schema.org/Product">
  <data itemprop="priceCurrency" value="EUR">€</data>
</div>

~~~~

The shorthand version fails 
~~~~
<div itemscope itemtype="http://schema.org/Product">
   <data itemprop="priceCurrency" value="EUR" />
</div>
~~~~

**Comment by AndySky21** [comment]
Please keep it in a single answer.

> The shorthand version fails

The shorthand version is your invention. The `<data>` element is not an empty element. See HTML spec for details.
`<data value="1234.56">1'234,56</data>`
In "value" goes the machine-readable value. Inside the element, the user-friendly string. No "shorthand" allowed, it isn't an empty element like `<meta>`

> This passes the W3 validator

Of course it does, it's valid HTML. But [Schema.org validator](https://validator.schema.org/) fails to parse the content of `value` attribute, violating Microdata spec and essentially making `<data>` useless.
And Google rich snippets parser fails too.
Interpreting `<data value=…>` is a fundamental part of Microdata, not different from `<a href=…>`

> The two online validators are not the same.

Same original project, different ports.

> This passes on W3.org and fails on Validator.nu

Some elements and attributes are not recognized the same because they are listed in W3C spec but not in WHATWG spec.
Actually this is not the case with `content=`, it isn't part of either specification. Just a bug, for what I know.

> I've known about WC3 for twenty some years, never heard of validator.nu

What can I say? Welcome to the jungle.
I have your very same background and I had a very bad time accepting the "politics" behind HTML and related tools.
W3 "validator.w3.org/NU" is called this way exactly because it embeds a port of Validator.NU "under the hood", adjusted to be consistent with **now obsolete** W3 HTML 5.2 spec.

**Comment by WeaverStever** [comment]
I was going to provide a working sample for the schema.org example, but the microdata is not ready for prime-time.

I'm testing on the Rich Snippets Testing Tool. The microdata "time" tag can be empty, but the "data" tag cannot.

Here are the hoops we will currently have to jump through to use non-visible "data" in microdata for Rich Snippets. 
~~~~
 <data itemprop="gtin13" value="1234567890123" style="display:none;">1234567890123</data>
~~~~
Completely unacceptable to have to display the value and then turn the display off with CSS.


Here is a working sample for JSON-LD (Rich Snippets Testing Tool)

~~~~
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "Product",
  "brand": "Kenmore",
  "sku": "abc-123",
  "gtin13": "1234567890123",
  "aggregateRating": {
    "@type": "AggregateRating",
    "ratingValue": "3.5",
    "reviewCount": "11"
  },
  "description": "0.7 cubic feet countertop microwave. Has six preset cooking categories and convenience features like Add-A-Minute and Child Lock.",
  "name": "Kenmore White 17\" Microwave",
  "image": "kenmore-microwave-17in.jpg",
  "offers": {
    "@type": "Offer",
    "url": "//example.com",
    "priceValidUntil": "2021-08-15",
    "availability": "https://schema.org/InStock",
    "price": "55.00",
    "priceCurrency": "USD"
  },
  "review": [
    {
      "@type": "Review",
      "author": "Ellie",
      "datePublished": "2011-04-01",
      "reviewBody": "The lamp burned out and now I have to replace it.",
      "name": "Not a happy camper",
      "reviewRating": {
        "@type": "Rating",
        "bestRating": "5",
        "ratingValue": "1",
        "worstRating": "1"
      }
    },
    {
      "@type": "Review",
      "author": "Lucas",
      "datePublished": "2011-03-25",
      "reviewBody": "Great microwave for the price. It is small and fits in my apartment.",
      "name": "Value purchase",
      "reviewRating": {
        "@type": "Rating",
        "bestRating": "5",
        "ratingValue": "4",
        "worstRating": "1"
      }
    }
  ]
}
</script>
~~~~
I don't know RDFa, nor have a desire to learn it -- dropping microdata may also be in my future. 
I'm also not interested in using any other testing tool, Google is 87% of my market.

**Comment by AndySky21** [comment]
@WeaverStever a recap.

- `<time>` is for date, period and time related **visible, user-friendly, natural language** expressions. Its `datetime` attribute handles machine-readable (ISO datetime) equivalent. Microdata Is supposed to consume `datetime`.
- `<meter>` is for scale values, **rendered visually for the user** in form of a color-coded bar. Its `value` attribute determines the bar fulfillment. Microdata is supposed to consume `value`, while other parameters must be expressed non-visually (see below)
- `<data>` is for any other case where a  **visible, user-friendly, natural language** expression must have machine-readable equivalent. Machine-readable equivalent is expressed by `value` attribute. Microdata consumes `value`.
- <meta> is for all those cases where there is **no visible, natural language** expression for the user. `<meta>` with an `itemprop` (Microdata) or a `property` (RDFa) attribute **can** be in _document body_. Its information is expressed by `content` attribute, which is consumed by either Microdata (if `itemprop` is also present) or RDFa (if `property` is also present).
- for other use cases, please read the specs. Good people have died (not quite literally) to write them.

There is no specific way to mark up other parameters for a meter scenario. Maximum value can be marked up via accompanying `<meta>` elements, while if a different data scale is to be expressed, it may be nested inside a `<data>` element.

**Comment by WeaverStever** [comment]
@AndySky21 

Here's a theory, because it seems to work everywhere, I wonder if the content= is not a bug, but rather a hack?

"Content" is so fundamental to html, this makes me wonder if "content" is the name of the handle for what lies between the opening and closing tags? I.e.` <div> content </div>
`
Consider this...
~~~~
<div itemprop="description">
   <p>My very long winded passage about my 1964 green Chevy.....</p>
</div>
~~~~

and this...
~~~~
<div itemprop="description" content="1964 green Chevy..">
   <p>My very long winded passage about my 1964 green Chevy.....</p>
</div>
~~~~
The content= hack overrides what the visible content, giving the structured data consumer a hint as to the subject of the passage, but the display remains the same for the reader.

This could also be useful where content is dynamic and the initial spider does not unpack the javascript.
~~~~
<div itemprop="description" content="1964 green Chevy..">
   <script>My very long winded passage about my 1964 green Chevy.....</script>
</div>
~~~~

So if "content" is the named variable for what lies between the opening and closing tags, it would be inadvisable to deprecate content=, rather use it as a hint as to why the passage has been included, and what the subject is.  The standards committees could then concern themselves as to which itemprop(s) are allowed to be empty. I.e., forcing some data elements to have "content" between the opening and closing tags so they are not placed as keyword spam.

***************************************

Interesting, the following fails Rich Snippets Testing Tool,
~~~~
<div itemscope itemtype="https://schema.org/Product">
  ...
  <data itemprop="sku" value="abc-123"></data>
  ...
  </div>
~~~~

while this succeeds 
~~~~
<div itemscope itemtype="https://schema.org/Product">
  ...
  <data itemprop="sku" value="abc-123" content="abc-123"></data>
  ...
  </div>
~~~~

this also passes the RSTT
~~~~
<div itemscope itemtype="https://schema.org/Product">
  ...
  <data itemprop="sku" value="abc-123" content="abc-123" />
  ...
  </div>
~~~~

**Comment by AndySky21** [comment]
@WeaverStever 
It's a lot longer and a lot sillier than this.
TL;DR: content is valid **HTML5** by a misunderstanding, but no longer valid **HTML** by literal intepretation and politics (also connected to that misunderstanding).

<details>
<summary>Click to dive into the Nightmare</summary>

I filed an [issue on Validator](https://github.com/validator/validator/issues/1210) regarding the different handling of the attribute content. Thats a recap, including what I was answered for that.
As stated before, `content` is part of RDFa. Now, there is a W3 specification called "HTML+RDFa 1.1" about implementation of RDFa in HTML5, in its 2 flavors: _core_ and _lite_. Because of a really poor phrasing of [Section 4 - Extensions to the HTML5 Syntax](https://www.w3.org/TR/html-rdfa/#extensions-to-the-html5-syntax), spec authors have had the brilliant idea of stating:

> If HTML+RDFa document conformance is desired, all RDFa attributes and valid values (including CURIEs), as listed in RDFa Core 1.1, Section 2.1: The RDFa Attributes, must be allowed and validate as conforming when used in an HTML4, HTML5 or XHTML5 document. For the avoidance of doubt, the following RDFa attributes are allowed on all elements in the HTML5 content model: `vocab`, `typeof`, `property`, `resource`, `prefix`, **`content`**, `about`, `rel`, `rev`, `datatype`, and `inlist`.

The spec should've been clearer and state: _the `content` attribute should be only present along with a consistent RDFa syntax in the same subtree_. But it isn't clear in doing that, so what we're left of is **HTML5** validator (_validator.w3.org/nu_, which is always a port of _validator.nu_ as stated before) considering valid `content` pretty much everywhere. 
I don't agree with that, probably some other people do not, but Validator maintainer has preferred to abide by the letter of this misunderstanding and grant generic support for RDFa attribute (including `content`).
In **HTML5**.

Now, as you probably know, W3 HTML had reached version 5.2, becoming Recommendation in 2017. But then W3C dropped the effort of producing further versions of **HTML5** (HTML 5.3 Working Draft has been reduced to Retired Note) in favor of electing as Recommendation [WHATWG HTML Standard](https://html.spec.whatwg.org/). WHATWG is the Working Group responsible for inventing all the shiny things _we call HTML 5_ today, including but not limited to Semantic Markup, Form Validation, Media Embedding, Web Workers and much more. In so doing, W3 has recomposed a long-time gap between those who _develop_ HTML<s>5</s> and those who officialize HTML5. More or less. 

The fact is, WHATWG no longer calls its standard HTML5. Their language is something free of versioning, in continuos evolution, and as such it is called **HTML Living Standard**.

**Previsions from [HTML+RDFa 1.1: Section 4 - Extensions to the HTML5 Syntax](https://www.w3.org/TR/html-rdfa/#extensions-to-the-html5-syntax) do not apply to HTML Living Standard, which is entitled to not consider as valid markup `content` (or any other RDFa attribute).** 
validator.nu, the official validating tool for WHATWG HTML Living Standard, acts as such.

There is, in fact, no spec related to integrating HTML LS and RDFa. And there will probably never be, because HTML Living Standard includes the spec for Microdata. We already know what Microdata states about machine-readable values. So, any need at all for `content`?
Not at all, **as far as Microdata-consuming tools interpret Rich Snippets correctly.**
Spoiler: they still don't, probably because until 28 January 2021 the relevant HTML spec was W3 HTML5, with its stupid mistake about `content` which nobody needed to care. One can only hope that Structured Data tools are still in process of upgrading.

Sorry for pontificating.

God help us.
</details>

**Comment by WeaverStever** [comment]
Sounds like this should be treated like a crypto-currency hard fork, let the population decide which platform they will use.

**Comment by chaals** [comment]
On Wed, 18 Aug 2021 05:40:04 +1000, Andrea Rendine  
***@***.***> wrote:

> any need at all for content?
> Not at all, as far as Microdata-consuming tools interpret Rich Snippets  
> correctly.
> Spoiler: they still don't, probably because until 28 January 2021 the  
> relevant HTML spec was W3 HTML5, with its stupid mistake >about content  
> which nobody needed to care. One can only hope that Structured Data  
> tools are still in process of upgrading.
(It was 2018 not 2021 that W3C work stopped)

They don't, because they were reading several specs badly. The schema.org  
examples were built based on the reality of implementations.

While W3C was still developing an HTML spec, there was also a concerted  
effort to make a microdata spec that matched reality - and as part of  
that, work with the various consumers of microdata including search  
engines and other processors, to update the tools so they actually worked  
the same way as the spec and vice versa.

As part of leaving it all to WHATWG, that spec (which was by then fairly  
accurate) was dumped in favour of the WHATWG version that was wildly out  
of touch with reality. I have no idea where it is now, but last I looked,  
maybe a year ago, they still had the same old stuff.

The W3C spec is available at  
https://www.w3.org/TR/2021/NOTE-microdata-20210128/

(I did a lot of work to get that microdata spec to happen and be  
worthwhile. I'm sad that we've gone so far backwards after that).

-- 
Using Opera's mail client: http://www.opera.com/mail/

**Comment by AndySky21** [comment]
> (It was 2018 not 2021 that W3C work stopped) 

Sorry, i was referencing the official date when W3 HTML 5.2 was officially superseded by WHATWG HTML LS.

> While W3C was still developing an HTML spec, there was also a concerted effort to make a microdata spec that matched reality - and as part of that, work with the various consumers of microdata including search engines and other processors, to update the tools so they actually worked the same way as the spec and vice versa. 

You guys reeeeally loved to read the Green Sentence _"W3 keeps publishing parts of WHATWG perfect work, although WHATWG Jedi council has officially prohibited them doing so"_ (not a 100% matching quote but that was the sense)

> As part of leaving it all to WHATWG, that spec (which was by then fairly accurate) was dumped in favour of the WHATWG version that was wildly out of touch with reality. 

I read zillions of article web-wide written by _bona fide_ charlatans, stating "RDFa rulez ovah Microdata bcs RDFa is Recommendation while Microdata is Working Draft", accurate or not. This has not helped gaining due traction, possibly the opposite.

> I have no idea where it is now, but last I looked, maybe a year ago, they still had the same old stuff. 

Well, now we're left with a new spec, which is Bible over HTML, Microdata and several other matters, even if we don't want it.

The point is, implementations might have followed a flawed standard, but now standards have changed. And among the countless **new** flaws introduced, one of the previous mistakes has been fixed (by accident). Indeed the `<data value="">` model is fairly elegant (more so than an ubiquitous `content` attribute) and I'm sure there are authors out there willing to write documents with Structured Data and also HTML compliant, which is now no longer possible. Implementations are not written on the stone.

The alternative is WHATWG HTML implementing `content` at least on `<data>` as an alternative to `value`. Which is asinine but maybe more conforming to reality.

@chaals you know better than any other what way is more likely to produce results. This is a minor but relevant issue, what should be done? Is it necessary to fetch data about use in the real world?
